packages/read-emails/frontend/react/src/EmailDetail.jsx

```diff
@@ -90,112 +167,169 @@ function EmailDetail({ selectedEmail, userEmail, serverBaseUrl, userId }) {
+                          <div className="attachment-files">
+                            {message.files
+                              .filter(
+                                (file) =>
+                                  file.content_disposition === 'attachment' &&
+                                  !file.content_type.includes('calendar')
+                              )
+                              .map((f) => (
+                                <div
+                                  className="attachment"
+                                  key={f.id}
+                                  onClick={(event) =>
+                                    downloadAttachment(event, f)
+                                  }
+                                >
+                                  <img
+                                    src={AttachmentIcon}
+                                    alt="attachment icon"
+                                    width="20"
+                                  />
+                                  <span>{f.filename}</span>
+                                </div>
+                              ))}
+                          </div>
+                        </div>
+                      )}
                     </div>
-                  </>
-                )}
-              </div>
-            )}
-          </div>
-          <div className="email-body">
-            <div
-              className="email-body-html"
-              dangerouslySetInnerHTML={{
-                __html: cleanEmailBody(selectedEmail.body),
-              }}
-            />
-            {!!attachments.length && (
-              <div className="attachment-container">
-                <div className="attachment-title">
-                  <span>Attachments</span>
-                  <hr />
-                </div>
+                  ) : (
+                    <p className="snippet">{message.snippet}</p>
+                  )}
 
-                <div className="attachment-files">
-                  {attachments.map((f) => (
+                  {isLoading && (
+                    <div className="loading-icon">
+                      <IconSync /> Loading...
+                    </div>
+                  )}
+
+                  {index !== messages.length - 1 && (
                     <div
-                      className="attachment"
-                      key={f.id}
-                      onClick={() => downloadAttachment(f)}
+                      className="message-border"
+                      onClick={handleShowCollapsedMessages}
                     >
-                      <img
-                        src={AttachmentIcon}
-                        alt="attachment icon"
-                        width="20"
-                      />
-                      <span>{f.filename}</span>
+                      {index === 0 && collapsedCount > 0 && (
+                        <span>Show {collapsedCount} messages</span>
+                      )}
```

Comment:
This should probably be hidden or say "hide x messages" when expanded

Reply:
This is hidden when `collapsedCount` has a value.

For the hide message option, since we don't have a design for that and Gmail also doesn't have the option to hide message once expanded, I think we can leave it.